analyzer/src/main/kotlin/managers/Pipenv.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer.managers
+
+import com.here.ort.analyzer.AbstractPackageManagerFactory
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.ProcessCapture
+import com.here.ort.utils.log
+import com.vdurmont.semver4j.Requirement
+import java.io.File
+
+object PipenvCli : CommandLineTool {
+
+    override fun command(workingDir: File?) = "pipenv"
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildIvy("[2018.10.9,)")
+
+    override fun getVersion(versionArguments: String, workingDir: File?, transform: (String) -> String) =
+        super.getVersion(versionArguments, workingDir) {
+            // "pipenv --version" returns a string like "pipenv, version 2018.11.26", so simply remove the prefix.
+            it.substringAfter("pipenv, version ")
+        }
+}
+
+class Pipenv(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : Pip(name, analysisRoot, analyzerConfig, repoConfig) {
+
+    class Factory : AbstractPackageManagerFactory<Pipenv>("Pipenv") {
+        override val globsForDefinitionFiles = listOf("Pipfile.lock")
```

Comment:
What if there is only a `Pipfile` but no `Pipfile.lock`? Couldn't we still generate the `requirements.txt`? In this case you would have to call `requireLockfile()` to throw create an error message if the `--allow-dynamic-versions` option is not set, similar to how it's done in other package managers like `Bundler` or `Npm`.

Reply:
This should not be the case normally. AFAIK `Pipenv` always assumes the generation of the lock-file and that this file should always be committed to your repo. So that Python env is always fully repeatable. That's why I specifically require lock-file being present, not the `Pipfile`

Reply:
That makes sense. If it turns out there are real life projects that do not commit their lock-file we can still easily adapt the implementation.